File: web-monitor/scripts/remove_duplicates.py
import sqlite3
import os

app_data = os.getenv('LOCALAPPDATA')
db_path = os.path.join(app_data, 'ProfileSubmissionAssistant', 'data.db')

conn = sqlite3.connect(db_path)
cursor = conn.cursor()

cursor.execute('SELECT Id, SiteName FROM Sites ORDER BY Id ASC')
sites = cursor.fetchall()

print(f"Total sites in DB: {len(sites)}")

seen_names = set()
duplicates = []

for site in sites:
    site_id = site[0]
    site_name = site[1]
    if site_name in seen_names:
        duplicates.append(site_id)
    else:
        seen_names.add(site_name)

print(f"Found {len(duplicates)} duplicate domains.")

if duplicates:
    for site_id in duplicates:
        cursor.execute('DELETE FROM SiteTasks WHERE SiteId = ?', (site_id,))
        cursor.execute('DELETE FROM Sites WHERE Id = ?', (site_id,))
    conn.commit()
    print("Duplicates removed.")
else:
    print("No duplicates found.")

conn.close()

File: web-monitor/scripts/remove_duplicates.js
import { queryDb } from '../src/lib/db';

async function removeDuplicates() {
  try {
    const sites = await queryDb('SELECT Id, SiteName, SignupUrl FROM Sites ORDER BY Id ASC');
    console.log(`Total sites in DB: ${sites.length}`);
    
    const seenNames = new Set();
    const duplicates = [];
    
    for (const s of sites) {
      if (seenNames.has(s.SiteName)) {
        duplicates.push(s.Id);
      } else {
        seenNames.add(s.SiteName);
      }
    }
    
    console.log(`Found ${duplicates.length} duplicate domains.`);
    
    if (duplicates.length > 0) {
      // Chunk duplicates deletion
      for (const id of duplicates) {
        await queryDb('DELETE FROM SiteTasks WHERE SiteId = ?', [id]);
        await queryDb('DELETE FROM Sites WHERE Id = ?', [id]);
      }
      console.log('Duplicates removed.');
    } else {
      console.log('No duplicates found.');
    }
    
  } catch (e) {
    console.error('Error removing duplicates', e);
  }
}

removeDuplicates();
